src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactorBackupCodes.tsx

```diff
@@ -97,21 +104,32 @@ export const SettingsEditSettingsTwoFactorBackupCodes: FC<SettingsEditSettingsTw
               <Spacer ml={1} />
 
               <Button
-                onClick={handleGenerate}
+                onClick={() => setShowConfirmPassword(true)}
                 loading={mode === "Creating"}
                 width={["100%", "auto"]}
               >
                 Regenerate
               </Button>
             </>
           ) : (
-            <Button onClick={handleGenerate} loading={mode === "Creating"}>
+            <Button
+              onClick={() => setShowConfirmPassword(true)}
+              loading={mode === "Creating"}
+            >
               Set up
             </Button>
           )}
         </Flex>
       </Flex>
 
+      <ConfirmPasswordModal
+        show={showConfirmPassword}
+        onConfirm={handleGenerate}
+        onCancel={() => setShowConfirmPassword(false)}
+        title="Regenerate Backup Codes"
```

Comment:
I chose to call this "Regenerate" seeing as whenever the user (initially) sets up 2fa, backup codes are automatically generated for them. In fact, I am curious if there is ever a state when a user will see "Set up" backup codes button?

Reply:
Nope. We conditionally [hide](https://github.com/artsy/force/blob/49e09ce13c0b6f8fd62794ebf65e85b45340c094/src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/SettingsEditSettingsTwoFactor.tsx#L33) the backup second-factor component unless the user has any form of 2fa, so they are only ever able to regenerate codes after setting up some other form of 2fa. This is by design, as we don't necessarily want users to rely on backup codes as their main form of 2fa.